tests/plugins/test_ddp_plugin_with_comm_hook.py

```diff
@@ -0,0 +1,112 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+from pytorch_lightning import Trainer
+from pytorch_lightning.plugins import DDPPlugin
+from pytorch_lightning.trainer.states import TrainerState
+from pytorch_lightning.utilities import _TORCH_GREATER_EQUAL_1_8
+from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+
+if torch.distributed.is_available() and _TORCH_GREATER_EQUAL_1_8:
+    from torch.distributed.algorithms.ddp_comm_hooks import (
+        default_hooks as default,
+        powerSGD_hook as powerSGD,
+    )
+
+
+@RunIf(skip_windows=True, min_torch="1.8.0", min_gpus=2)
```

Comment:
```suggestion
@RunIf(skip_windows=True, min_torch="1.8.0", min_gpus=2, special=True)
```

Reply:
Make sure you test for `ddp_spawn` and `dpp`. Trainer(accelerator="ddp"). `ddp` tests requieres `special=True` flag.